src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -167,34 +166,18 @@ private static String extractQualifiedName(DetailAST classExtend) {
     }
 
     /**
-     * Register to outer super classes of given classAst that
-     * given classAst is extending them.
-     * @param classAst class which outer super classes will be
-     *                 informed about nesting subclass
-     */
-    private void registerNestedSubclassToOuterSuperClasses(DetailAST classAst) {
-        final String currentAstSuperClassName = getSuperClassName(classAst);
-        if (currentAstSuperClassName != null) {
-            for (ClassDesc classDesc : classes) {
-                final String classDescQualifiedName = classDesc.getQualifiedName();
-                if (doesNameInExtendMatchSuperClassName(classDescQualifiedName,
-                        currentAstSuperClassName)) {
-                    classDesc.registerNestedSubclass();
-                }
-            }
-        }
-    }
-
-    /**
      * Get qualified class name from given class Ast.
      * @param classAst class to get qualified class name
      * @return qualified class name of a class
      */
     private String getQualifiedClassName(DetailAST classAst) {
         final String className = classAst.findFirstToken(TokenTypes.IDENT).getText();
         String outerClassQualifiedName = null;
-        if (!classes.isEmpty()) {
-            outerClassQualifiedName = classes.peek().getQualifiedName();
+        for (ClassDesc desc : classes) {
```

Comment:
Can we use Stream API here?
1) Filter by predicate
2) Map the filter result to the string (use ```getQualifiedClassName``` as the lambda for the map)